Added pull_request_template.md

danielchen-pyc opened the pull request:
# Changes
- Added pull request template in the `.github` folder. 
- @johnagl 

# Checklist

danielchen-pyc committed:
Added pull_request_template.md

jacksonyuanjx commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,9 @@
+- Change summary/description
+- Any @mention 's
+
+# Related Tickets
```
Should we rename this as "Related Issues" since I don't think we're using tickets in Launch Pad? Please correct me if I'm wrong @johnagl

danielchen-pyc replied:
I copied this from the launchpad doc repo's PR template and they were using `tickets` as the title but actually using `closing issues#` in the content. I thought they are the same but I not sure. @johnagl

johnagl replied:
Issues is probably a bit clearer, but up to you if you want to change @danielchen-pyc :)

jacksonyuanjx commented:
What do you guys think of additional sections such as "How has this been tested?", "Motivation/Context", "Screenshots (if applicable)", etc.?

I found this online and it seems to have some good options too: https://github.com/stevemao/github-issue-templates/blob/master/questions-answers/PULL_REQUEST_TEMPLATE.md

johnagl commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,9 @@
+- Change summary/description
+- Any @mention 's
+
+# Related Tickets
```
It's not too big of a deal, but I guess Issues would be clearer, up to you if you want to change @danielchen-pyc

danielchen-pyc committed:
Added some descriptive titles. Changed tickets->issues

danielchen-pyc commented:
Since this PR hasn't been merged yet, I changed some features regarding the PR template. Below is the new PR template (and PR also loll). 

# Changes
- Changed tickets to issues
- Added the three sections below for more descriptive explanation of the changes. 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

# Related Issues
- Closes issue #

# Checklist

johnagl reviewed (approved):
Awesome @danielchen-pyc!

danielchen-pyc commented:
@johnagl @jacksonyuanjx Hello! I'm still getting familiar with this issue workflow. I was wondering if what I did was correct. So basically, after this PR was approved, I closed issue #7 since PR template has been created and approved. I merge `daniel` branch to `master` and safely deleted the temporary `daniel` branch. Did I do anything wrong? Thank you!!

jacksonyuanjx commented:
> @johnagl @jacksonyuanjx Hello! I'm still getting familiar with this issue workflow. I was wondering if what I did was correct. So basically, after this PR was approved, I closed issue #7 since PR template has been created and approved. I merge `daniel` branch to `master` and safely deleted the temporary `daniel` branch. Did I do anything wrong? Thank you!!

Hi! What you described sounds good to me, but I'll let John have the final say :)
